R/string_operation.R

```diff
@@ -909,6 +909,62 @@ str_remove_emoji <- function(column, position = "any"){
   stringi::stri_replace_all(column, regex = regexp, "")
 }
 
+#'Function to extract text before the separator.
+#'
+#'export
+str_extract_before <- function(column, sep = "\\,", include_sep = TRUE) {
+  if (include_sep) {
+    stringr::str_extract(column, stringr::str_c(".*", sep))
+  } else {
+    stringr::str_extract(column, stringr::str_c("(.*)(?=", sep, ")"))
+  }
+}
+
+#'Function to extract text after the separator.
+#'
+#'export
+str_extract_after <- function(column, sep = "\\,", include_sep = TRUE){
+  if (include_sep){
+    stringr::str_extract(column, stringr::str_c(sep, ".*"))
+  } else {
+    stringr::str_extract(column, stringr::str_c("(?<=", sep, ")(.*)"))
+  }
+}
+#'Function to replace text before the separator.
+#'
+#'export
+str_replace_before <- function(column, sep = ",", rep = "", include_sep = TRUE) {
```

Comment:
I think whichever would work, but only this function's sep is not escaped by `\\`.

Reply:
added `\\` for consistency.